### PR TITLE
CRUD for applications

### PR DESCRIPTION
**Pull request by pottekkat**
**Description**

This PR adds CRUD operations on the Meshery Server for "applications".

Support for this on meshery-cloud has been added in [this PR](https://github.com/layer5io/meshery-cloud/pull/81).

**Notes for Reviewers**

@Chadha93 There is an issue with decoding the request body when creating uploading a new application file.

Could you check if there are any issues with the request being sent from the Meshery UI?

This request is handled in `line 52` in `meshery_application_handler.go` file-

```go
	var parsedBody *MesheryApplicationRequestBody
	if err := json.NewDecoder(r.Body).Decode(&parsedBody); err != nil {
```

Basically, the parsed data is empty. I don't see any issues on the Meshery Server. To be sure, please check the UI.

So, when I try to upload a file, the UI throws this error-

`invalid applicationfile - name field is either not present or is not valid`

Screenshot_20210616_103656

This is thrown from `line 39` in `models/meshery_application.go`.

// @utkarsh-pro 

- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@utkarsh-pro  the error documented in the Reviewer’s notes happens when the remote provider’s capabilities.json doesn’t support the resource type, is that right?

**Comment by leecalcote** [comment]
@Chadha93 what do you think of the changes in this PR?

**Comment by Chadha93** [comment]
> @Chadha93 what do you think of the changes in this PR?

I didn't find any issues with the UI and this also seems fine to me but as @navendu-pottekkat said it not properly working (CRUD) ......

**Comment by leecalcote** [comment]
> @utkarsh-pro the error documented in the Reviewer’s notes happens when the remote provider’s capabilities.json doesn’t support the resource type, is that right?

@utkarsh-pro will you put eyeballs on this PR for review when you have some time?

**Comment by utkarsh-pro** [comment]
> > @utkarsh-pro the error documented in the Reviewer’s notes happens when the remote provider’s capabilities.json doesn’t support the resource type, is that right?
> 
> @utkarsh-pro will you put eyeballs on this PR for review when you have some time?

Ah apologies, I missed this comment @leecalcote. @Chadha93 I think you have a typo https://github.com/navendu-pottekkat/meshery/blob/feature%2Fnavendu-pottekkat%2Fapplications/ui/components/MesheryApplications.js#L219 here which is causing the error.

**Comment by Chadha93** [comment]
> > > @utkarsh-pro the error documented in the Reviewer’s notes happens when the remote provider’s capabilities.json doesn’t support the resource type, is that right?
> > 
> > 
> > @utkarsh-pro will you put eyeballs on this PR for review when you have some time?
> 
> Ah apologies, I missed this comment @leecalcote. @Chadha93 I think you have a typo https://github.com/navendu-pottekkat/meshery/blob/feature%2Fnavendu-pottekkat%2Fapplications/ui/components/MesheryApplications.js#L219 here which is causing the error.

@utkarsh-pro Thanks for this, I forgot to add  `_`  in `applicationfile` @navendu-pottekkat should I create a new pr with changes?

**Comment by navendu-pottekkat** [comment]
@Chadha93 I have fixed the UI bug and tested it out. Looks good.

Thank you @leecalcote @utkarsh-pro 

Screenshot_20210623_233515

Please review and merge the changes.

**Review by leecalcote — APPROVED** [review]
